File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/Google/Person.cs
using System.Collections.Generic;
using System.Text;

public class Person
{
    public Person(string name)
    {
        this.Name = name;
        this.Company = new Company();
        this.Car = new Car();
        this.Pokemons = new List<Pokemon>();
        this.Parents = new List<Relative>();
        this.Children = new List<Relative>();
    }

    public string Name { get; set; }
    public Company Company { get; set; }
    public Car Car { get; set; }
    public List<Pokemon> Pokemons { get; set; }
    public List<Relative> Parents { get; set; }
    public List<Relative> Children { get; set; }

    public override string ToString()
    {
        var sb = new StringBuilder();
        sb.AppendLine(this.Name);
        sb.AppendLine($"Company:");
        if (this.Company.Name != null)
        {
            sb.AppendLine(this.Company.ToString());
        }

        sb.AppendLine($"Car:");

        if (this.Car.Model != null)
        {
            sb.AppendLine(this.Car.ToString());
        }

        sb.AppendLine($"Pokemon:");
        foreach (var pokemon in this.Pokemons)
        {
            sb.AppendLine(pokemon.ToString());
        }

        sb.AppendLine($"Parents:");
        foreach (var parent in this.Parents)
        {
            sb.AppendLine(parent.ToString());
        }

        sb.AppendLine($"Children:");
        foreach (var child in this.Children)
        {
            sb.AppendLine(child.ToString());
        }

        return sb.ToString().Trim();
    }
}

File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Exercise/pr09_google/Person.java
package Lesson01DefiningClasses.Exercise.pr09_google;

import java.util.LinkedHashSet;
import java.util.Set;

public class Person {
    private String name;
    private Company company;
    private Car car;
    private Set<FamilyMember> parents;
    private Set<FamilyMember> children;
    private Set<Pokemon> pokemonList;

    public Person(String name) {
        super();
        this.name = name;
        this.parents = new LinkedHashSet<>();
        this.children = new LinkedHashSet<>();
        this.pokemonList = new LinkedHashSet<>();
    }

    void setCompany(Company company) {
        this.company = company;
    }

    void setCar(Car car) {
        this.car = car;
    }

    void addParent(FamilyMember parent) {
        this.parents.add(parent);
    }

    void addChild(FamilyMember child) {
        this.children.add(child);
    }

    void addPokemon(Pokemon pokemon) {
        this.pokemonList.add(pokemon);
    }

    String getInfo() {
        StringBuilder sb = new StringBuilder();
        sb.append(this.name).append(System.lineSeparator());
        sb.append("Company:").append(System.lineSeparator());
        if (this.company != null) {
            sb.append(this.company.getInfo()).append(System.lineSeparator());
        }
        sb.append("Car:").append(System.lineSeparator());
        if (this.car != null) {
            sb.append(this.car.getInfo()).append(System.lineSeparator());
        }
        sb.append("Pokemon:").append(System.lineSeparator());
        if (!this.pokemonList.isEmpty()) {
            for (Pokemon pokemon : this.pokemonList) {
                sb.append(pokemon.getInfo()).append(System.lineSeparator());
            }
        }
        sb.append("Parents:").append(System.lineSeparator());
        if (!this.parents.isEmpty()) {
            for (FamilyMember parent : this.parents) {
                sb.append(parent.getInfo()).append(System.lineSeparator());
            }
        }
        sb.append("Children:").append(System.lineSeparator());
        if (!this.children.isEmpty()) {
            for (FamilyMember child : this.children) {
                sb.append(child.getInfo()).append(System.lineSeparator());
            }
        }
        return sb.toString();
    }
}
